src/sentry/api/endpoints/organization_recent_searches.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.serializers import serialize
+from sentry.models.recentsearch import RecentSearch
+from sentry.models.search_common import SearchType
+
+
+class OrganizationRecentSearchesEndpoint(OrganizationEndpoint):
+
+    def get(self, request, organization):
+        """
+        List recent searches for a User within an Organization
+        `````````````````````````````````````
+        Returns recent searches for a user in a given Organization.
+
+        :auth: required
+
+        """
+        search_type = SearchType(int(request.GET.get('type', 0)))
```

Comment:
Should we accept a partial query string as a parameter? Or are we going to 'autocomplete' recent searches in the client?

Reply:
My plan was to follow the `query` part up in a separate diff, although @billyvg mentioned he might prefer to just handle it on the frontend. I'm happy to go with either option.

Reply:
I think adding it in a follow up PR would be fine.

Reply:
That works for me.